### PR TITLE
Added simple C# example

### PR DESCRIPTION
**Pull request by CurtHagenlocher**

### CODE REVIEW

File: http/get_simple/csharp/client/ArrowHttpClient.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            {
+                BaseAddress = new Uri(serverUri),
+            };
+
+            using (var stream = await httpClient.GetStreamAsync(serverUri))
+            using (var reader = new ArrowStreamReader(stream))
+            {
+                Console.WriteLine("Connected");
+
+                Schema schema = reader.Schema;
```
Comment (ianmcook) [inline http/get_simple/csharp/client/ArrowHttpClient.cs:45]: This line doesn't seem to be setting the schema. If I try to access it after, I get a `NullReferenceException`.

**Comment by ianmcook** [comment]
Thanks @CurtHagenlocher!

I tested these with all other clients and servers, and everything works, except that I needed to set the server URI to `"http://localhost:8008/"` or `"http://*:8008/"` to get some of the client/server pairs to work.

**Comment by CurtHagenlocher** [comment]
I confess I only tested with the Rust implementation, and at least on my Windows box the Rust client did not work with localhost so I changed it to the IP address.

File: http/get_simple/csharp/client/ArrowHttpClient.cs (hunk 1 above)
Comment (CurtHagenlocher) [reply http/get_simple/csharp/client/ArrowHttpClient.cs:45]: This looks like a bug in the C# IPC reader where reader.Schema isn't set until after the first batch is read. I filed [#40634](https://github.com/apache/arrow/issues/40634) to track this.

Comment (ianmcook) [reply http/get_simple/csharp/client/ArrowHttpClient.cs:45]: Thanks. In the meantime: I moved this line lower in #26.

**Comment by ianmcook** [comment]
Cool, thanks. I changed the server URIs in #26 for consistency / interoperability.

**Comment by ianmcook** [comment]
I tested a version of the C# client example that writes the received data to an IPC stream file ([Gist here](https://gist.github.com/ianmcook/7c9e4033b59bf71245603b67e69e4235)) and it works as expected.

**Pull request merged**